docs/appendix/glossary/modular-resource.md

```diff
@@ -0,0 +1,11 @@
+---
+title: Modular Resource
+id: modular-resource
+full_link:
+short_description: A modular resource is a custom component, service, or logic implemented through Viam's public APIs.
+aka:
+---
+
+A modular resource is a custom {{< glossary_tooltip term_id="component" text="component" >}}, {{< glossary_tooltip term_id="service" text="service" >}}, or logic implemented through [Viam's public APIs](/extend/modular-resources/key-concepts/#valid-apis-to-implement-in-your-model).
```

Comment:
We do have another definition here which could also use a bit of work but I think the following suggestion works:  

```suggestion
A modular resource is a custom {{< glossary_tooltip term_id="model" text="model" >}} of an existing {{< glossary_tooltip term_id="component" text="component" >}} or {{< glossary_tooltip term_id="service" text="service" >}} that implements the same [API methods](https://docs.viam.com/program/apis/).
```